reputation/reputation.go

```diff
@@ -0,0 +1,413 @@
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for rows.Next() {
+		var name string
+		var timestamp string
+		var uptime int
+		var auditSuccess int
+		var auditFail int
+		var latency int
+		var amountOfDataStored int
+		var falseClaims int
+		var shardsModified int
+
+		err = rows.Scan(
+			&name,
+			&timestamp,
+			&uptime,
+			&auditSuccess,
+			&auditFail,
+			&latency,
+			&amountOfDataStored,
+			&falseClaims,
+			&shardsModified,
+		)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		currentRow := RepRow{
+			name,
+			timestamp,
+			uptime,
+			auditSuccess,
+			auditFail,
+			latency,
+			amountOfDataStored,
+			falseClaims,
+			shardsModified,
+		}
+
+		res = append(res, currentRow)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer rows.Close()
+
+	return res
+}
+
+// close sqlite3
+func cleanUpDB(db *sql.DB) {
+	defer db.Close()
+}
+
+// finds the ration of audit success from the success and failure fields of a given reputaion row struct
+func (row RepRow) auditSuccessRatio() float64 {
+	return float64(row.auditSuccess) / float64(row.auditSuccess+row.auditFail)
+}
+
+// naive formula for obtaining repuataion
+func (row RepRow) naiveRep() float64 {
+	var mutator int
+
+	if row.shardsModified > 0 {
+		mutator = 0
+	} else {
+		mutator = 1
+	}
+
+	return (float64(row.uptime*100) +
+		row.auditSuccessRatio() +
+		float64(row.latency) +
+		float64(row.amountOfDataStored) -
+		float64(row.falseClaims)) * float64(mutator)
+}
+
+// compares reputation rows and returns the greater reputation of the two
+func (row RepRow) greaterRep(other RepRow) RepRow {
+	myRep := row.naiveRep()
+	otherRep := other.naiveRep()
+	myTime := row.timestamp
+	otherTime := other.timestamp
+
+	var res RepRow
+
+	switch {
+	case myTime < otherTime:
+		res = other
+	case myTime > otherTime:
+		res = row
+	case myRep > otherRep:
+		res = row
+	default:
+		res = other
+	}
+
+	return res
+}
+
+// finds the naive reputation of the resulting rows from the query string
+func naiveReputation(db *sql.DB, queryString string) RepRow {
+	bestRep := RepRow{"identity", "", 0, 0, 0, 0, 0, 0, 0}
+
+	rows, err := db.Query(queryString)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for rows.Next() {
+		var name string
+		var timestamp string
+		var uptime int
+		var auditSuccess int
+		var auditFail int
+		var latency int
+		var amountOfDataStored int
+		var falseClaims int
+		var shardsModified int
+
+		err = rows.Scan(
+			&name,
+			&timestamp,
+			&uptime,
+			&auditSuccess,
+			&auditFail,
+			&latency,
+			&amountOfDataStored,
+			&falseClaims,
+			&shardsModified)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		currentRow := RepRow{
+			name,
+			timestamp,
+			uptime,
+			auditSuccess,
+			auditFail,
+			latency,
+			amountOfDataStored,
+			falseClaims,
+			shardsModified}
+
+		bestRep = bestRep.greaterRep(currentRow)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer rows.Close()
+
+	return bestRep
+}
+
+type column string
```

Comment:
I think we should split this file into a couple of different logical files. This seems like a good place to separate.